connection_pool/connection_pool.go

```diff
@@ -32,17 +32,31 @@ var (
 	ErrNoHealthyInstance = errors.New("can't find healthy instance in pool")
 )
 
-/*
-Additional options (configurable via ConnectWithOpts):
-
-- CheckTimeout - time interval to check for connection timeout and try to switch connection.
-*/
+// ConnectionHandler provides callbacks for components interested in handling
+// changes of connections in a ConnectionPool.
+type ConnectionHandler interface {
+	// Opened is called when a new connection has been opened. It has not yet
+	// been added to the pool of active connections. It allows for a client to
+	// initialize the connection before using it in a pool.
+	//
+	// The client code may cancel adding a connection to the pool. The client
+	// need to return an error from the Opened call for that. In this case
+	// the pool will close connection and will try to reopen it later.
+	Opened(conn *tarantool.Connection, role Role) error
+	// Closed is called after a connection with a role removed from a list
+	// of active connections. If a role of the connection is updated, a
```

Comment:
`Opened` description should also contain the information that it is called if a role of the connection is updated